explainer.md

```diff
@@ -0,0 +1,329 @@
+
+In the following sections we'll explore each of these major categories of functionality
+and how we propose the accessibility API would work.
+
+### Exploring the accessibility tree
+
+The most basic thing that's possible using the proposed Accessibility Object Model
+is exploring and querying the existing accessibility tree.
+
+With the AOM, every DOM node has a property **`accessibleNode`**
+that accesses that element's corresponding object in the accessibility tree,
+if it has one.
+Most DOM nodes will have an accessible node, with a few exceptions -
+an element might not have an accessible node if it's not currently attached to a visible Document,
+or if it's not currently displayed, for example.
+
+Accessing an element's accessible node allows you to determine its role.
+This example shows how you could figure out the role assigned to an HTML INPUT element with a type of "range":
+
+```html
+<label>
+  Rating:
+  <input id="myinput" type="range" value="5" min="1" max="10">
+</label>
+<script>
+  var input = document.querySelector("#myinput");
+  var axInput = input.accessibleNode;
+  axInput.role;  // returns "slider"
+</script>
+
+```
+
+Most ARIA attributes have a corresponding property on an accessible node.
+In this particular case, we can access
+the min, max, and current value of the slider
+and its text label, among other things.
+
+```js
+axInput.rangeValue;  // returns 5.0
+axInput.rangeMin;    // returns 1.0
+axInput.rangeMax;    // returns 10.0
+axInput.label;       // returns "Rating:"
+```
+
+
+## Use cases
+
+### Use DOM node references instead of IDREFs
+Some HTML and ARIA features require reference to another element by IDREF,
+but it's problematic in some contexts to require the accessibility code to generate IDs.
+For example, some JS frameworks generated IDs automatically
+and make maintaining the value of those IDs purposefully obtuse.
+Likewise, in some extremes cases,
+the expectation of IDREFs requires generating thousands of DOM modifications
+which can lead to serious performance problems in the application.
+
+Moreover, IDREFs cannot cross over Shadow DOM boundaries,
+so any case which requires an ARIA relationship across these boundaries cannot be implemented today
+
+This API would allow authors to associate related elements using a JavaScript object reference.
+
+#### Example code
+
+```html
+<x-combobox>
+  <x-optionlist>
+    <x-option>Option 1</x-option>
+    <x-option>Option 2</x-option>
+    <x-option>Option 3</x-option>
+ </x-optionlist>
+</x-combobox>
+```
+
+```js
+var combobox = document.querySelector('x-combobox');
+var shadowRoot = combobox.createShadowRoot();
+var input = document.createElement('input');
+shadowRoot.appendChild(input);
+
+var option1 = combobox.querySelectorAll('x-option')[0];
+var axInput = input.accessibleNode;
+var axOption1 = option1.accessibleNode;
+axInput.activeDescendant = axOption1;
+```
+
+### Create virtual accessibility trees
```

Comment:
Would I be able to tab through a virtual accessibility tree or would it only be accessible via screen reader?